worlds/jakanddaxter/regs/GolAndMaiasCitadelRegions.py

```diff
@@ -0,0 +1,130 @@
+from typing import List
+from BaseClasses import CollectionState, MultiWorld
+from .RegionBase import JakAndDaxterRegion
+from .. import JakAndDaxterOptions, EnableOrbsanity
+from ..Rules import can_free_scout_flies, can_fight, can_reach_orbs
+
+
+# God help me... here we go.
+def build_regions(level_name: str, multiworld: MultiWorld, options: JakAndDaxterOptions, player: int) -> List[JakAndDaxterRegion]:
+
+    # This level is full of short-medium gaps that cannot be crossed by single jump alone.
+    # These helper functions list out the moves that can cross all these gaps (painting with a broad brush but...)
+    def can_jump_farther(state: CollectionState, p: int) -> bool:
```

Comment:
The first two lines of `can_jump_further` and `can_jump_stairs` could use `state.has_any`